### PR TITLE
Checkout tests

### PR DESCRIPTION
**Pull request by ostapneko**
The plan is to add unit tests to 
- [x] `IdentityService`
- ~~[ ] `SalesforceService`~~

And then mock these services to test
- [x] `CheckoutService`

### CODE REVIEW

File: app/services/IdentityService.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@ import play.api.libs.ws.{WS, WSRequest, WSResponse}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging {
+class IdentityService(identityApiClient: => IdentityApiClient) extends LazyLogging {
```
Comment (afiore) [inline app/services/IdentityService.scala]: Is there a reason why we are passing in `IdentityApiClient` by name?

File: test/services/CheckoutServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      def process(service: CheckoutService): Future[CheckoutResult] =
+        service.processSubscription(subscriptionData, None, None)
+
+      val checkoutResult = Await.result(
+        process(service),
+        2.seconds
+      )
+
+      "does not update the user details in the Identity service" in {
+        var updated = false
```
Comment (afiore) [inline test/services/CheckoutServiceSpec.scala]: Checking for the effect by mutating a var is not great, but I guess we have no better way of doing it...

**Comment by afiore** [comment]
:+1: Looking good!

File: app/services/IdentityService.scala (hunk 1 above)
Comment (ostapneko) [inline app/services/IdentityService.scala]: To be able to replace it by `???` in the tests

File: test/services/CheckoutServiceSpec.scala (hunk 2 above)
Comment (ostapneko) [inline test/services/CheckoutServiceSpec.scala]: I'm open to alternatives

**Pull request merged**